src/plugins/data_source_management/public/components/data_source_menu/data_source_menu.tsx

```diff
@@ -4,94 +4,68 @@
  */
 
 import React, { ReactElement } from 'react';
-import { EuiHeaderLinks } from '@elastic/eui';
-import classNames from 'classnames';
 
-import {
-  MountPoint,
-  NotificationsStart,
-  SavedObjectsClientContract,
-  SavedObject,
-} from '../../../../../core/public';
-import { MountPointPortal } from '../../../../opensearch_dashboards_react/public';
 import { DataSourceSelectable } from './data_source_selectable';
-import { DataSourceOption } from '../data_source_selector/data_source_selector';
 import { DataSourceAggregatedView } from '../data_source_aggregated_view';
 import { DataSourceView } from '../data_source_view';
-import { DataSourceAttributes } from '../../types';
-
-export interface DataSourceMenuProps {
-  showDataSourceSelectable?: boolean;
-  showDataSourceView?: boolean;
-  showDataSourceAggregatedView?: boolean;
-  activeDataSourceIds?: string[];
-  appName: string;
-  savedObjects?: SavedObjectsClientContract;
-  notifications?: NotificationsStart;
-  fullWidth: boolean;
-  hideLocalCluster?: boolean;
-  dataSourceCallBackFunc?: (dataSource: DataSourceOption) => void;
-  disableDataSourceSelectable?: boolean;
-  className?: string;
-  selectedOption?: DataSourceOption[];
-  setMenuMountPoint?: (menuMount: MountPoint | undefined) => void;
-  dataSourceFilter?: (dataSource: SavedObject<DataSourceAttributes>) => boolean;
-  displayAllCompatibleDataSources?: boolean;
-}
-
-export function DataSourceMenu(props: DataSourceMenuProps): ReactElement | null {
-  const {
-    savedObjects,
-    notifications,
-    dataSourceCallBackFunc,
-    showDataSourceSelectable,
-    disableDataSourceSelectable,
-    showDataSourceAggregatedView,
-    fullWidth,
-    hideLocalCluster,
-    selectedOption,
-    showDataSourceView,
-    dataSourceFilter,
-    activeDataSourceIds,
-    displayAllCompatibleDataSources,
-  } = props;
+import {
+  DataSourceAggregatedViewConfig,
+  DataSourceComponentType,
+  DataSourceMenuProps,
+  DataSourceSelectableConfig,
+  DataSourceViewConfig,
+} from './types';
 
-  if (!showDataSourceSelectable && !showDataSourceView && !showDataSourceAggregatedView) {
-    return null;
-  }
+export function DataSourceMenu<T>(props: DataSourceMenuProps<T>): ReactElement | null {
+  const { componentType, componentConfig } = props;
```

Comment:
Any reason why we cannot simply made the type prop be `type`?
Also, is there a reason to have a bunch of props clubbed into `config`? To me, it feels natural not club them.

Reply:
@AMoo-Miki , this was based on discussion with @ashwin-pc @Flyingliuhub @ZilongX  @yujin-emma last week to group the configs specific to a component and use generic type for better readability which I think makes a lot of sense here. If we don't do that, all the configs to all components will need to the the prop for Data Source Menu. Also, this is similar to how TopNavMenu accepts the config https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/navigation/public/top_nav_menu/top_nav_menu.tsx#L47